src/components/topics.jsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from "react";
+import { getAllTopics, selectArticlesByTopics } from "../api";
+import { Link } from "react-router-dom";
+import { useParams } from "react-router-dom";
+import ArticleList from "./articleList";
+
+const Topics = () => {
+  const [topics, setTopics] = useState([]);
+  const [articles, setArticles] = useState([]);
+  const { topic_name } = useParams();
+  console.log(topic_name);
+  useEffect(() => {
+    getAllTopics().then((topics) => {
+      setTopics(topics.topics);
+    });
+
+    // if (!topic_name) {
+    //   console.log("no topic");
+    //   selectArticlesByTopics().then((allTopics) => {
+    //     console.log(allTopics);
+    //     setArticles(allTopics.articles);
+    //   });
+    // } else {
+    selectArticlesByTopics(topic_name).then((topic) => {
```

Comment:
I would like us to rename 'topic' here to be something a bit more semantic since this is not a topic object, it is an articles object. To make this cleaner i would suggest `{articles}` to destructure articles and make it more semantic, and then change the setArticles arguement below to just `articles`